docs/design/cronjob.md

```diff
@@ -0,0 +1,118 @@
+    // - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+    // - "Replace": cancels currently running job and replaces it with a new one
+    // +optional
+    ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty" protobuf:"bytes,4,opt,name=concurrencyPolicy,casttype=ConcurrencyPolicy"`
+
+    // This flag tells the controller to suspend subsequent executions, it does
+    // not apply to already started executions.  Defaults to false.
+    // +optional
+    Suspend *bool `json:"suspend,omitempty" protobuf:"varint,5,opt,name=suspend"`
+    
+    // The number of successful finished jobs to retain.
+    // This is a pointer to distinguish between explicit zero and not specified.
+    // Defaults to 3.
+    // +optional
+    SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"varint,6,opt,name=successfulJobsHistoryLimit"`
+
+    // The number of failed finished jobs to retain.
+    // This is a pointer to distinguish between explicit zero and not specified.
+    // Defaults to 1.
+    // +optional
+    FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty" protobuf:"varint,7,opt,name=failedJobsHistoryLimit"`
+}
+
+// ConcurrencyPolicy describes how the job will be handled.
+// Only one of the following concurrent policies may be specified.
+// If none of the following policies is specified, the default one
+// is AllowConcurrent.
+type ConcurrencyPolicy string
+
+const (
+	// AllowConcurrent allows CronJobs to run concurrently.
+	AllowConcurrent ConcurrencyPolicy = "Allow"
+
+	// ForbidConcurrent forbids concurrent runs, skipping next run if previous
+	// hasn't finished yet.
+	ForbidConcurrent ConcurrencyPolicy = "Forbid"
+
+	// ReplaceConcurrent cancels currently running job and replaces it with a new one.
+	ReplaceConcurrent ConcurrencyPolicy = "Replace"
+)
+
+// CronJobStatus represents the current state of a cron job.
+type CronJobStatus struct {
+	// A list of pointers to currently running jobs.
+	// +optional
+	Active []v1.ObjectReference `json:"active,omitempty" protobuf:"bytes,1,rep,name=active"`
+
+	// Information when was the last time the job was successfully scheduled.
+	// +optional
+	LastScheduleTime *metav1.Time `json:"lastScheduleTime,omitempty" protobuf:"bytes,4,opt,name=lastScheduleTime"`
+}
+
+```
+
+### CronJob Controller
+
+The `CronJobController` will run periodically, watching all `CronJob` resource as well as their batch job resources.
+It's responsible for creating/deleting batch job resource according to `CronJobSpec` and current state, also updating
+the state of `CronJob`.
+
+### Admission Controller
+
+The admission controller will ensure the correctness of `JobSpec`, especially the `Schedule` expression.
+
+### Feature Interaction
+
+#### cli
+
+Command line support for `CronJob` is not considered at this first stage, However end user still can manipulate 
```

Comment:
we need to design it here, and finish it in different release.

Reply:
ok.